telegram/telegram.py

```
@@ -0,0 +1,176 @@
+        self.outbound_url = config.outbound_url + config.bot_token
+        self.inbound_url = config.inbound_url
+        self.bot_username = config.bot_username
+        self.web_path = config.web_path
+        self.web_port = config.web_port
+        self.clock = self.get_clock()
+        self.rpc_resource = HttpRpcResource(self)
+
+        self.web_resource = yield self.start_web_resources(
+            [
+                (self.rpc_resource, self.web_path),
+            ],
+            self.web_port)
+
+        yield self.setup_webhook()
+
+    def get_clock(self):
+        return reactor
+
+    @inlineCallbacks
+    def handle_raw_inbound_message(self, message_id, request):
+        update = json.loads(request.content.read())
+        update_id = update['update_id']
+
+        # Telegram updates can contain objects other than messages (ignore if
+        # that is the case)
+        if 'message' not in update:
+            log.info('Inbound update does not contain a message')
+            return
+
+        message = self.translate_inbound_message(update['message'])
+        log.info(
+            'TelegramTransport receiving inbound message from %s to %s' % (
+                message['from_addr'], message['to_addr']))
+
+        request.finish()
+        yield self.publish_message(
+            message_id=message_id,
+            content=message['content'],
+            to_addr=message['to_addr'],
+            from_addr=message['from_addr'],
+            transport_type=self.transport_type,
+            transport_name=self.transport_name,
+        )
+
+    def translate_inbound_message(self, message):
+        """
+        Translates inbound Telegram message into Vumi's preferred format
+        """
+        # We are only interested in text messages for now
+        if 'text' in message:
+            content = message['text']
+        else:
+            log.info('Message is not a text message.')
+            content = ''
```

I think this should count as an error case and we should just log something and throw away the message rather than processing it as an empty message.